src/validation.ts

```diff
@@ -0,0 +1,126 @@
+import Ajv, { AnySchema, DefinedError } from 'ajv';
+import addFormats from 'ajv-formats';
+
+import { AssertionError } from './error';
+import { Assert, Guard } from './interfaces';
+
+export function preparePrepareIs(
+  schema: AnySchema
+): <T>(id: string) => Guard<T> {
+  const ajv = new Ajv({ allowUnionTypes: true });
```

Comment:
I was thinking about: https://ajv.js.org/guide/managing-schemas.html#compiling-during-initialization - wouldn't be more efficient to to create only one Ajv instance before running this function (like after Ajv import ) 🤷🏻‍♂️. Also we should  decide if we want to use addSchema or compile - it is possible that both will be similarly effective- it is just about when we are compiling the scheme. 

I am not against current solution - but if we are trying to lessen the memory load we should look into this stuff :)

Reply:
I was sure that you won't like it 😂 This looks ok (I mean effective).